packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -818,6 +821,25 @@ export function $clientName(
   value: string,
   scope?: LanguageScopes
 ) {
+  if (entity.kind === "Model" || entity.kind === "Operation") {
```

Comment:
Can you add a reference to this issue https://github.com/microsoft/typespec/issues/2717

that this is a workaround for the lack of functionality in the compiler.